gpflow/kullback_leiblers.py

```diff
@@ -71,32 +73,45 @@ def gauss_kl(q_mu, q_sqrt, K=None, *, K_cholesky=None):
     is_white = (K is None) and (K_cholesky is None)
     is_diag = len(q_sqrt.shape) == 2
 
-    M, B = tf.shape(q_mu)[0], tf.shape(q_mu)[1]
+    shape_constraints = [
+        (q_mu, ["M", "L"]),
+        (q_sqrt, (["M", "L"] if is_diag else ["L", "M", "M"])),
```

Comment:
above you've used `"ML"` etc instead of `["M", "L"]` etc. do we want to stick with one?